### DISCUSSION TITLE
How to set idle time for lock-screen

### DISCUSSION THREAD

**Discussion by YerongLi** [i3/i3#6172] [Ideas]
I don't want i3 constantly lock screen on idle.

`xset s 7200` in the terminal or `exec --no-startup-id xset s 7200` in the `i3/config` do not work.

Even though `xset q` output some changes.
```
xset q
Screen Saver:
  prefer blanking:  yes    allow exposures:  yes
  timeout:  7200    cycle:  0
```
Are there any setting GUI in i3?

**Comment by okraits** [comment]
i3 is a window manager. It only manages windows and workspaces, nothing else.

Have a look at https://wiki.archlinux.org/title/Display_Power_Management_Signaling#Runtime_settings
Note that this affects only screen blanking and monitor power management, not screen locking.

If your screen gets locked then something in the background triggers screen locking after a certain idle time.
Have a look at https://wiki.archlinux.org/title/Session_lock

**Reply by YerongLi** [reply]
Tested below, and general xset seems not working, you do know which option should work, I want to prevent frequent logout.

```
xset s off -dpms
xset s
xset -dpms
xset s 3600 3600
xset s 3600 3600
```

```
# Set timeout to 2 hours
exec --no-startup-id xset s 7200
exec --no-startup-id xset s 7200 7200
```

**Reply by YerongLi** [reply]
In XCFE4, I was able to set the lock-out time to 9999 seconds. I would preassume that no lockup triggers exist.
> If your screen gets locked then something in the background triggers screen locking after a certain idle time. Have a look at https://wiki.archlinux.org/title/Session_lock

**Comment by YerongLi** [comment]
Still no luck with `xset`, Not sure whether there are other solutions to try
